From eb5a757e3ad4f096b17084cd7782eb52d7833760 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 7 Oct 2022 12:00:39 +0200
Subject: [PATCH 0164/1695] clk: stm32mp13: add CLK_SET_RATE_PARENT &
 CLK_OPS_PARENT_ENABLE flags

These flags are needed to change frequency of the MPU clock.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3d7fa75c4623db0b5303aa87c633ce400858ddce clk: stm32mp13: add CLK_SET_RATE_PARENT & CLK_OPS_PARENT_ENABLE flags]
Change-Id: I0dd96cd9d6a9c914623cdf82fd41b4aabe27073b
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index 646d8b8c2..1891ca714 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -2012,7 +2012,7 @@ static struct clk ck_pll1p = {
 		.mux_id		= NO_MUX,
 	},
 	.name		= "ck_pll1p",
-	.flags		= 0,
+	.flags		= CLK_SET_RATE_PARENT,
 	.num_parents	= 1,
 	.parents	= { &ck_pll1_vco },
 };
@@ -2027,7 +2027,7 @@ static struct clk ck_pll1p_div = {
 		.div_id	= DIV_MPU,
 	},
 	.name	= "ck_pll1p_div",
-	.flags	= 0,
+	.flags	= CLK_SET_RATE_PARENT,
 	.num_parents	= 1,
 	.parents	= { &ck_pll1p },
 };
@@ -2078,7 +2078,7 @@ static struct clk ck_mpu = {
 		.mux_id	= MUX_MPU,
 	},
 	.name		= "ck_mpu",
-	.flags		= 0,
+	.flags		= CLK_OPS_PARENT_ENABLE | CLK_SET_RATE_PARENT,
 	.num_parents	= 4,
 	.parents	= { &ck_hsi, &ck_hse, &ck_pll1p, &ck_pll1p_div },
 };
-- 
2.47.2

